polynote-frontend/polynote/ui/display/display_content.ts

```diff
@@ -6,16 +6,109 @@ import {Content, details, div, span, tag, TagElement} from "../tags";
 import {ArrayType, MapType, OptionalType, StructField, StructType} from "../../data/data_type";
 import embed from "vega-embed";
 
+const frames: Record<number, HTMLIFrameElement> = {};
+let nextFrameId = 1;
+
+function getNextFrameId() {
+    return nextFrameId++;
+}
+
+// listen for sizing events from the frames
+window.addEventListener('message', evt => {
+   if (evt.data.command && evt.data.command === 'size' && evt.data.id && frames[evt.data.id]) {
+       const frame = frames[evt.data.id];
+       if (frame.offsetWidth !== evt.data.height) {
+           frame.style.height = evt.data.height + 'px';
+       }
+       // if (frame.offsetWidth !== evt.data.width) {
+       //     frame.style.width = evt.data.width + 'px';
+       // }
+   }
+});
+
+let resizeTimeout: number;
+// recheck sizes when the document resizes
+window.addEventListener('resize', evt => {
+    if (resizeTimeout) {
+        window.clearTimeout(resizeTimeout);
+    }
+    resizeTimeout = window.setTimeout(updateSizes, 250);
+})
+
+function updateSizes() {
```

Comment:
Does this still work if the frame is currently out of view (when another notebook tab is selected)?

Reply:
I think so? I mean the function will work, but I don't know what the frame will say about itself if its host `iframe` isn't in the DOM. 😬